### PR TITLE
[17.11] Fix component governance alerts

### PR DESCRIPTION
**Pull request by MichalPavlik**
Fixes ~~CVE-2024-38081~~, CVE-2024-38095

### Context
Some of our dependencies contains vulnerabilities. 

### Changes Made
I backported changes we already have in `main` branch - updated ~~`Microsoft.IO.Redist`~~ package version and pinned `System.Formats.Asn1` package version.

### Testing
Existing unit test.

### Notes
VS 17.11 still uses `Microsoft.IO.Redist` version 6.0.0, so we need to stick with this version.

### CODE REVIEW

**Comment by JanKrivanek** [comment]
FYI @marcpopMSFT

**Review by JanKrivanek — APPROVED** [review]

File: eng/Versions.props [hunk 1]
```diff
@@ -50,7 +51,7 @@
     <MicrosoftCodeAnalysisCollectionsVersion>4.2.0-1.22102.8</MicrosoftCodeAnalysisCollectionsVersion>
     <MicrosoftDotNetXUnitExtensionsVersion>8.0.0-beta.24311.3</MicrosoftDotNetXUnitExtensionsVersion>
     <MicrosoftExtensionsDependencyModelVersion>7.0.0</MicrosoftExtensionsDependencyModelVersion>
-    <MicrosoftIORedistVersion>6.0.0</MicrosoftIORedistVersion>
+    <MicrosoftIORedistVersion>6.0.1</MicrosoftIORedistVersion>
```
Comment (rainersigwald) [inline eng/Versions.props:54]: This required a coordinated change with VS; are you planning to drive that for 17.11?

File: src/MSBuild/app.amd64.config [hunk 2]
```diff
@@ -49,8 +49,8 @@
 
         <dependentAssembly>
           <assemblyIdentity name="Microsoft.IO.Redist" culture="neutral" publicKeyToken="cc7b13ffcd2ddd51" />
-          <bindingRedirect oldVersion="0.0.0.0-99.9.9.9" newVersion="6.0.0.0" />
-          <codeBase version="6.0.0.0" href="..\Microsoft.IO.Redist.dll"/>
+          <bindingRedirect oldVersion="0.0.0.0-99.9.9.9" newVersion="6.0.0.1" />
```
Comment (rainersigwald) [inline src/MSBuild/app.amd64.config:52]: This was wrong before; we only get to use the huge range on our own assemblies

```suggestion
          <bindingRedirect oldVersion="0.0.0.0-6.0.0.1" newVersion="6.0.0.1" />
```

File: eng/Versions.props (hunk 1 above)
Comment (MichalPavlik) [reply eng/Versions.props:54]: I will contact QB for VS and ask if the change is required for VS. It's possible we will not insert to VS. There could be a problem when we will have to backport something else that will be required to be inserted.

**Comment by MichalPavlik** [comment]
@GangWang01, could you please check what's wrong with the check? I bumped the VersionPrefix servicing number, but it still complains.

File: eng/Versions.props (hunk 1 above)
Comment (MichalPavlik) [reply eng/Versions.props]: As discussed offline, we will not update the MS.IO.Redist package version

File: src/MSBuild/app.amd64.config [hunk 3]
```diff
@@ -49,7 +49,7 @@
 
         <dependentAssembly>
           <assemblyIdentity name="Microsoft.IO.Redist" culture="neutral" publicKeyToken="cc7b13ffcd2ddd51" />
-          <bindingRedirect oldVersion="0.0.0.0-99.9.9.9" newVersion="6.0.0.0" />
+          <bindingRedirect oldVersion="0.0.0.0-6.0.0.1" newVersion="6.0.0.0" />
```
Comment (maridematte) [inline src/MSBuild/app.amd64.config:52]: There seems like the version was not reverted as it was before.

**Review by maridematte — CHANGES_REQUESTED** [review]

File: src/MSBuild/app.amd64.config (hunk 3 above)
Comment (MichalPavlik) [reply src/MSBuild/app.amd64.config:52]: `newVersion` is the version which image loader picks. `oldVersion` represents a range for assembly that is requested. I believe it's still valid.

File: src/MSBuild/app.config [hunk 4]
```diff
@@ -43,6 +43,10 @@
           <assemblyIdentity name="Microsoft.Bcl.AsyncInterfaces" publicKeyToken="cc7b13ffcd2ddd51" culture="neutral" />
           <bindingRedirect oldVersion="0.0.0.0-8.0.0.0" newVersion="8.0.0.0" />
         </dependentAssembly>
+        <dependentAssembly>
+          <assemblyIdentity name="Microsoft.IO.Redist" culture="neutral" publicKeyToken="cc7b13ffcd2ddd51" />
+          <bindingRedirect oldVersion="0.0.0.0-6.0.0.1" newVersion="6.0.0.0" />
+        </dependentAssembly>
```
[the remark this answers is not in the document]
Comment (MichalPavlik) [reply src/MSBuild/app.config:49]: Same explanation as before :)

**Review by maridematte — APPROVED** [review]

**Pull request merged**